payroll-processor-functions/src/payroll-processor-functions/Features/Payrolls/PayrollTrigger.cs

```diff
@@ -0,0 +1,156 @@
+                payroll.EmployeeDepartment.ToLowerInvariant(),
+                payroll.EmployeeId.ToString("n"),
+                EmployeeEntity.Map.To);
+
+            var employee = option.IfNone(() => throw new Exception($"Could not find employee [{payroll.EmployeeDepartment}] [{payroll.EmployeeId}]"));
+
+            var tableResult = await payrollsTable.ExecuteAsync(TableOperation.Insert(PayrollEntity.Map.From(payroll)));
+
+            if (!(tableResult.Result is PayrollEntity payrollEntity))
+            {
+                throw new Exception($"Could not save payroll");
+            }
+
+            await payrollUpdatesQueue.AddMessageAsync(EntityQueueMessageProcessor.ToQueueMessage(payrollEntity));
+
+            payroll = PayrollEntity.Map.To(payrollEntity);
+
+            return new OkObjectResult(payroll);
+        }
+
+        [FunctionName("Payroll_Update")]
+        public async Task<IActionResult> UpdatePayroll(
+                [HttpTrigger(AuthorizationLevel.Anonymous, "PUT", Route = "payrolls")] HttpRequest req,
+                [Table("payrolls")] CloudTable payrollsTable,
+                [Queue("payroll-updates")] CloudQueue payrollUpdatesQueue,
+                ILogger log)
+        {
+            log.LogInformation($"Creating a new payroll: [{req}]");
+
+            var payroll = await Request.Parse<Payroll>(req);
+
+            var payrollEntity = PayrollEntity.Map.From(payroll);
+
+            var payrollUpdate = TableOperation.Replace(payrollEntity);
+
+            await payrollsTable.ExecuteAsync(payrollUpdate);
+
+            await payrollUpdatesQueue.AddMessageAsync(EntityQueueMessageProcessor.ToQueueMessage(payrollEntity));
+
+            return new OkObjectResult(payroll);
+        }
+
+        [FunctionName("PayrollQueue_Update")]
+        public async Task UpdatePayrollFromQueue(
+                [QueueTrigger("payroll-updates")] CloudQueueMessage message,
+                [Table("employees")] CloudTable employeeTable,
+                [Table("employeePayrolls")] CloudTable employeePayrollsTable,
+                ILogger log)
+        {
+            log.LogInformation($"Processing payrollUpdates queue: [{message}]");
+
+            var payrollEntity = EntityQueueMessageProcessor.FromQueueMessage<PayrollEntity>(message);
+
+            var payroll = PayrollEntity.Map.To(payrollEntity);
+
+            var querier = new TableQuerier(employeeTable);
+
+            var option = await querier.GetEntity<EmployeeEntity, Employee>(
+                payrollEntity.EmployeeDepartment.ToLowerInvariant(),
+                payrollEntity.EmployeeId.ToString("n"),
+                EmployeeEntity.Map.To);
+
+            var employee = option.IfNone(() => throw new Exception($"Could not find employee for payroll [{payroll.Id}]"));
+
+            employee
+                .Payrolls
+                .Find(p => p.Id == payroll.Id)
+                .Match(p =>
```

Comment:
The way this works is, if the payroll is found in the employee's `Payrolls` set, then update it.

Otherwise, insert it if the length of the employee's `Payrolls` is less than 30. If it's 30 or greater, don't make any changes.